lib/types.go

```diff
@@ -41,6 +42,18 @@ type Options struct {
 	Thresholds map[string][]*Threshold `json:"thresholds"`
 }
 
+type SourceData struct {
```

Comment:
Ah, I thought of it as the *source data type* rather than the runner type, and wanted to have a struct that contained as much information as possible about the source data supplied by the user. I.e. the actual data, any filename the data was read from, and what type of data it is. I guess the type is useless inside a runner, because the runner will only be started for one single type of data so it already knows what it is [supposed to be], but if we use the struct elsewhere, outside of the runner, maybe it could be useful to include that information?  I'll remove it if you think it doesn't make sense.